vars/cloudFoundryDeploy.groovy

```diff
@@ -118,8 +119,14 @@ def deployCfNative (config) {
         usernameVariable: 'username'
     )]) {
         def deployCommand = 'push'
+        def blueGreenDeployOptions = ''
+        boolean deleteOldInstance = !config.keepOldInstance
+
         if (config.deployType == 'blue-green') {
             deployCommand = 'blue-green-deploy'
+            if (deleteOldInstance) {
+                blueGreenDeployOptions = '--delete-old-apps'
```

Comment:
Are we certain this does not override any options defined before? Currently yes, but is this obvious once some one make a change here? Might the options better be a list, which then is transformed to a space separated string? (many examples for this in various steps here)

Reply:
exported the calculation of the commands into own methods